syntax = "proto3";

package ed;

import "typedef.proto";

// heartbeat
message Heartbeat
{
	int32 connection_id = 1;
}

message HeartbeatResp
{
	int32  error_code = 1;
    string error_msg = 2;
	string server_time = 3;
}

// 查询合约列表
message QueryAllInsInfo
{

}

message QuoteBasicInfo
{
	// 合约基本信息
	TypeDef.InstrumentType ins_type = 1; 	// 合约类型
	TypeDef.Exchange exchange = 2;			// 交易所ID
	string ins_number = 3;					// 合约序号
	string ins_id = 4;						// 合约ID
	string ins_name = 5;         			// 合约名称
	string ins_en_name = 6;        			// 合约英文名
	TypeDef.Currency settle_currency = 7;   // 结算货币
	TypeDef.Currency trade_currency = 8;	// 交易货币
	// 品种基本信息
	TypeDef.ProductType product_type = 9;  // 品种类型
    string product_code = 10;               // 品种代码
	string product_name = 11;               // 品种名称
	string product_en_name = 12;			// 品种英文名称
	string trading_time_slice = 13;         // 交易时间片

	int32 lot_size = 14;                  	// 每手数量
	double min_change_price_gap = 15;       // 最小变动价差
	bool is_trade = 16;                     // 是否可交易
	TypeDef.InstrumentStatus ins_status = 17; // 合约状态
	string create_date = 18;                  // 创建日期
	string listing_date = 19;				  // 上市日期 YYYYMMDD
	string update_date = 20;                  // 更新日期
	// 期货合约私有信息
	TypeDef.RateType deposit_collect_method = 21; // 保证金收取方式
	double long_deposit_rate = 22;            // 多头保证金率
	double short_deposit_rate = 23;           // 空头保证金率
	int32 market_price_max_vol = 24;          // 市价单最大数量
	int32 market_price_min_vol = 25;          // 市价单最小数量
	int32 limit_price_max_vol = 26;           // 限价单最大数量
	int32 limit_price_min_vol = 27;           // 限价单最小数量
	double ins_multiplier = 28;               // 合约乘数
	int32 delivery_year = 29;                 // 交割年
	int32 delivery_mouth = 30;                // 交割月
	bool is_expired = 31;                     // 是否过期
	bool is_main_ins = 32;                    // 是否是主力合约
	string trade_last_date = 33;              // 最后交易日
	string first_notice_date = 34;            // 首次通知日
	string expired_date = 35;                 // 到期日
	int32 expired_mouth = 36;                 // 到期月份
	string start_delivery_date = 37;          // 开始交割日期 YYYYMMDD
	string end_delivery_date = 38;            // 结束交割日期 YYYYMMDD
};

message QueryAllInsInfoResp
{
	int32  error_code = 1;
    string error_msg = 2;
	repeated QuoteBasicInfo data = 3;
}

// 订阅行情
message SubQuotes
{
	TypeDef.SubType type = 1;				// 订阅类型
	TypeDef.Exchange exchange = 2;			// 交易所ID
	string ins_id = 3;						// 合约ID
}

message SubQuotesResp
{
	int32  error_code = 1;
    string error_msg = 2;
}

// 取消订阅行情
message UnSubQuotes
{
	TypeDef.SubType type = 1;				// 订阅类型
	TypeDef.Exchange exchange = 2;			// 交易所ID
	string ins_id = 3;						// 合约ID
}

message UnSubQuotesResp
{
	int32  error_code = 1;
    string error_msg = 2;
}